packages/react-data-grid/src/HeaderCell.tsx

```diff
@@ -146,23 +147,16 @@ export default class HeaderCell<R> extends React.Component<HeaderCellProps<R>> {
         onMouseDown={column.resizable ? this.onMouseDown : undefined}
         onTouchStart={column.resizable ? this.onTouchStart : undefined}
       >
-        {this.getCell()}
+        {rowType === HeaderRowType.HEADER && column.draggable && DraggableHeaderCell ? (
```

Comment:
Move the `dragTarget` to cell inner level so that we don't see the issue in this link.

https://github.com/react-dnd/react-dnd/issues/1608